src/Common/src/TypeSystem/IL/ILImporter.cs

```diff
@@ -116,6 +116,33 @@ private BasicBlock CreateBasicBlock(int offset)
                 basicBlock = new BasicBlock() { StartOffset = offset };
                 _basicBlocks[offset] = basicBlock;
             }
+
```

Comment:
I see.. What about brute-force searching based on index? (still no fancy algorithm, but not slower than the old one) https://github.com/dotnet/corert/pull/3513/commits/6efb538094b2c275887c68e3214ebb0967fec409